worker/rhpv2.go

```diff
@@ -593,6 +595,69 @@ func (s *Session) Write(ctx context.Context, actions []rhpv2.RPCWriteAction, pri
 	return nil
 }
 
+func (s *Session) Reconnect(ctx context.Context, hostIP string, hostKey types.PublicKey, renterKey types.PrivateKey, contractID types.FileContractID) (err error) {
+	defer wrapErr(&err, "Reconnect")
+
+	if s.transport != nil {
+		s.transport.Close()
+	}
+
+	conn, err := (&net.Dialer{}).DialContext(ctx, "tcp", hostIP)
+	if err != nil {
+		return err
+	}
+	s.transport, err = rhpv2.NewRenterTransport(conn, hostKey)
+	if err != nil {
+		return err
+	}
+
+	s.key = renterKey
+	if err = s.Lock(ctx, contractID, renterKey, 10*time.Second); err != nil {
+		s.transport.Close()
+		return err
+	}
+
+	if err := s.updateSettings(ctx); err != nil {
+		s.transport.Close()
+		return err
+	}
+
+	s.lastSeen = time.Now()
+	return nil
+}
+
+func (s *Session) Refresh(ctx context.Context, sessionTTL time.Duration, renterKey types.PrivateKey, contractID types.FileContractID) error {
```

Comment:
Docstrings on `Refresh` vs `Reconnect` would be nice. Split up like this it's not quite as obvious anymore.

Reply:
There are docstrings on `Refresh` and `Reconnect`? Is this an outdated comment perhaps that doesn't make sense because of force pushing

Reply:
resolving, I think this is the case